stl/inc/expected

```diff
@@ -87,7 +87,11 @@ public:
     }
 
     friend constexpr void swap(unexpected& _Left, unexpected& _Right) noexcept(is_nothrow_swappable_v<_Err>)
+#if defined(__clang__) || defined(__EDG__) // TRANSITION, DevCom-10652420
         requires is_swappable_v<_Err>
+#else // ^^^ no workaround / workaround vvv
+        requires is_swappable<_Err>::value
+#endif // ^^ workaround ^^^

```

Comment:
It might be cleaner to create a macro like `_STL_TRAIT_PERMISSIVE` that does this which takes the name and dispatches accordingly:
```cpp
#if defined(__clang__) || defined(__EDG__) // TRANSITION, DevCom-10652420
#  define _STL_TRAIT_PERIMISSIVE(__TRAIT, ...) __TRAIT##_v<__VA_ARGS__>
#else // ^^^ no workaround / workaround vvv
#  define _STL_TRAIT_PERIMISSIVE(__TRAIT, ...) __TRAIT<__VA_ARGS__>::value
#endif // ^^ workaround ^^^
```

Reply:
It seems that only `is_swappable_v` needs the workaround (perhaps together with `is_swappable_with_v`, but the latter is unused in the STL) because of ADL. I'd like to consider this in a future PR which will change a C++23 `std::swap` overload for `pair`.

Reply:
Extensive macroization interferes with global code search, which dramatically amplifies my power. I'd like to avoid such macros in product code. If the verbosity increases, I'd rather just unconditionally use `::value` in affected locations with a comment `// TRANSITION, /permissive needs ::value` as a reminder. Emitting workarounds for compilers that don't need them is generally undesirable, but this case could be barely tolerable.